src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -929,6 +942,106 @@ private void DecodeModuleInitializerAttribute(DecodeWellKnownAttributeArguments<
             }
         }
 
+        private void DecodeInterceptsLocationAttribute(DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments)
+        {
+            Debug.Assert(arguments.AttributeSyntaxOpt is object);
+            Debug.Assert(!arguments.Attribute.HasErrors);
+            var attributeArguments = arguments.Attribute.CommonConstructorArguments;
+            if (attributeArguments is not [
+                { Kind: not TypedConstantKind.Array, Value: string filePath },
+                { Kind: not TypedConstantKind.Array, Value: int displayLineNumber },
+                { Kind: not TypedConstantKind.Array, Value: int displayCharacterNumber }])
+            {
+                // Since the attribute does not have errors (asserted above), it should be guaranteed that we have the above arguments.
+                throw ExceptionUtilities.Unreachable();
+            }
+
+            // Internally, line and character numbers are 0-indexed, but when they appear in code or diagnostic messages, they are 1-indexed.
+            int lineNumber = displayLineNumber - 1;
```

Comment:
Can the line number or character number be set to `0` or some negative values like `int.MinValue`?